lib/sps_king/account/creditor_address.rb

```diff
@@ -20,18 +20,47 @@ class CreditorAddress
     convert :address_line1,   to: :text
     convert :address_line2,   to: :text
 
-    validates_length_of :street_name,     maximum: 70
-    validates_length_of :building_number, maximum: 16
-    validates_length_of :post_code,       maximum: 16
-    validates_length_of :town_name,       maximum: 35
-    validates_length_of :country_code,    is: 2
-    validates_length_of :address_line1,   maximum: 70
-    validates_length_of :address_line2,   maximum: 70
+    validates :country_code,  presence: true, format: { with: /\A[A-Z]{2}\z/ }
+    validates :address_line1,                 length: { maximum: 70 }, if: :lines_any?
+    validates :address_line2, presence: true, length: { maximum: 70 }, if: :lines_any?
+    validates :post_code,     presence: true, length: { maximum: 16 }, if: :separate_any?
+    validates :town_name,     presence: true, length: { maximum: 35 }, if: :separate_any?
+    validates :street_name,                   length: { maximum: 70 }, if: :separate_any?
```

Comment:
@tobischo 

Removed the presence validation for address_line1 and street_name.

Reply:
BTW I'm just wondering:

Any reason why CreditorAddress has the attribute building_number and DebtorAddress does not?

Reply:
No reason - we probably just missed it and never noticed because we didn't use it